src/Stormpath/Resource/Tenant.php

```diff
@@ -71,6 +72,18 @@ public function getDirectories(array $options = array())
         return $this->getResourceProperty(self::DIRECTORIES, Stormpath::DIRECTORY_LIST, $options);
     }
 
+    public function getCustomData(array $options = array())
+    {
+        $customData =  $this->getResourceProperty(self::CUSTOM_DATA, Stormpath::CUSTOM_DATA, $options);
+
+        if(!$customData) {
+            $customData = new CustomData();
+            $this->setProperty(self::CUSTOM_DATA, $customData);
+        }
+
+        return $customData;
+    }
```

Comment:
The code of this function repeats on all resources so I think we should have a parent class that implements this logic.